src/System.Runtime/tests/System/Enum.cs

```diff
@@ -4,219 +4,165 @@
+    [InlineData(0xffffff80u, null)]
+    [InlineData(unchecked((int)(0xffffff80u)), "Min")]
+    [InlineData(true, "One")]
+    [InlineData((char)1, "One")]
+    [InlineData(SimpleEnum.Red, "One")] // API doesnt't care if you pass in a completely different enum
+    public static void TestGetNameNonItegralTypes(object value, string expected)
+    {
+        /*
+            * Despite what MSDN says, GetName() does not require passing in the exact integral type.
+            * 
+            * For the purposes of comparison:
+            * 
+            *  - The enum member value are normalized as follows:
+            *    - unsigned ints zero-extended to 64-bits
+            *    - signed ints sign-extended to 64-bits
+            *
+            *  - The value passed in as an argument to GetNames() is normalized as follows:
+            *    - unsigned ints zero-extended to 64-bits
+            *    - signed ints sign-extended to 64-bits
+            *
+            *  Then comparison is done on all 64 bits.
+            */
+        String s = Enum.GetName(typeof(SByteEnum), value);
+        Assert.Equal(expected, s);
+    }
+
+    [Theory]
+    [InlineData(typeof(SimpleEnum), "Red", true)] //String
+    [InlineData(typeof(SimpleEnum), "Green", true)]
+    [InlineData(typeof(SimpleEnum), "Blue", true)]
+    [InlineData(typeof(SimpleEnum), "blue", false)]
+    [InlineData(typeof(SimpleEnum), "", false)]
+    [InlineData(typeof(SimpleEnum), SimpleEnum.Red, true)] //Enum
+    [InlineData(typeof(SimpleEnum), (SimpleEnum)99, false)]
+    [InlineData(typeof(SimpleEnum), 1, true)] // Integer
+    [InlineData(typeof(SimpleEnum), 99, false)]
+    [InlineData(typeof(Int32Enum), 0x1 | 0x02, false)] // "Combos" do not pass
+    public static void TestIsDefined(Type t, object value, bool expected)
+    {
+        bool b = Enum.IsDefined(t, value);
+        Assert.Equal(expected, b);
+    }
 
-        // Other negative tests
-        Assert.Throws<ArgumentNullException>(() => Enum.IsDefined(null, 1));
-        Assert.Throws<ArgumentNullException>(() => Enum.IsDefined(t, null));
+    [Fact]
+    public static void TestIsDefinedInvalid()
+    {
+        Type t = typeof(SimpleEnum);
+        
+        Assert.Throws<ArgumentNullException>("enumType", () => Enum.IsDefined(null, 1)); // Enum type is null
+        Assert.Throws<ArgumentNullException>("value", () => Enum.IsDefined(t, null)); // Value is null
 
-        // These throws ArgumentException (though MSDN claims it should throw InvalidOperationException)
-        Assert.Throws<ArgumentException>(() => Enum.IsDefined(t, true));
-        Assert.Throws<ArgumentException>(() => Enum.IsDefined(t, 'a'));
+        Assert.Throws<ArgumentException>(null, () => Enum.IsDefined(t, Int32Enum.One)); // Value is different enum type
+                                                                                           
+        // Value is not a valid type (MSDN claims this should throw InvalidOperationException)
```

Comment:
@stephentoub 
MSDN is incorrect - it says that we should throw an InvalidOperationException. I suspect that the code for Enum is just checking to see if the type is the underlying enum type irrespective of whether it is a supported enum typeS
See here:
![isdefined msdn](https://cloud.githubusercontent.com/assets/1275900/11280434/f80a03a0-8eed-11e5-97a1-2681ca028a4a.png)
 https://msdn.microsoft.com/en-us/library/system.enum.isdefined%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396